File: api/Enkl.Api/Controllers/AnnouncementsController.cs
using Enkl.Api.Auth;
using Enkl.Api.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Enkl.Api.Controllers;

/// <summary>
/// Any authenticated user (no OrgAdmin/ProjectMember policy) — reading "what's currently relevant to
/// me" and acknowledging one you've seen are things every signed-in user can do, not an admin-only
/// action. See OrganisationAnnouncementsController for the Org-Admin-only CRUD management surface.
/// </summary>
[ApiController]
[Authorize]
[Route("api/announcements")]
public class AnnouncementsController : ControllerBase
{
    private readonly AnnouncementService _announcements;

    public AnnouncementsController(AnnouncementService announcements)
    {
        _announcements = announcements;
    }

    [HttpGet("active")]
    public async Task<IActionResult> GetActive()
    {
        return Ok(await _announcements.GetActiveForUserAsync(User.OrgId(), User.UserId()));
    }

    [HttpPost("{announcementId:guid}/acknowledge")]
    public async Task<IActionResult> Acknowledge(Guid announcementId)
    {
        await _announcements.AcknowledgeAsync(User.UserId(), announcementId);
        return NoContent();
    }
}

File: mariadb-api/src/Controllers/AnnouncementsController.php
<?php

declare(strict_types=1);

namespace Enkl\Api\Controllers;

use Enkl\Api\Db\Database;
use Enkl\Api\Services\AnnouncementService;
use Psr\Http\Message\ResponseInterface as Response;
use Psr\Http\Message\ServerRequestInterface as Request;

/**
 * Ported from php-api's Controllers/AnnouncementsController.php. Any authenticated user (no
 * OrgAdmin/ProjectMemberMiddleware) — reading "what's currently relevant to me" and acknowledging one
 * you've seen are things every signed-in user can do, not an admin-only action. See
 * OrganisationAnnouncementsController for the Org-Admin-only CRUD management surface.
 */
final class AnnouncementsController extends BaseController
{
    private function service(): AnnouncementService
    {
        return new AnnouncementService(Database::connection());
    }

    public function getActive(Request $request, Response $response): Response
    {
        return $this->json($response, $this->service()->getActiveForUser($this->callerOrgId($request), $this->callerUserId($request)));
    }

    public function acknowledge(Request $request, Response $response, array $args): Response
    {
        $this->service()->acknowledge($this->callerUserId($request), $args['announcementId']);
        return $this->noContent($response);
    }
}
